web-common/src/main/java/org/jboss/as/web/session/SharedSessionManagerConfig.java

```diff
@@ -31,25 +32,28 @@
  */
 public class SharedSessionManagerConfig {
 
-    public static final ServiceName SHARED_SESSION_MANAGER_SERVICE_NAME = ServiceName.of("undertow", "shared-session-manager");
+    public static final AttachmentKey<SharedSessionManagerConfig> ATTACHMENT_KEY = AttachmentKey.create(SharedSessionManagerConfig.class);
+    public static final ServiceName SHARED_SESSION_MANAGER_SERVICE_NAME = ServiceName.of("web", "shared-session-manager");
     public static final ServiceName SHARED_SESSION_IDENTIFIER_CODEC_SERVICE_NAME = SHARED_SESSION_MANAGER_SERVICE_NAME.append("codec");
 
-    private int maxActiveSessions = -1;
+    private Integer maxActiveSessions;
```

Comment:
Why the change to an object?

Reply:
2 reasons:
1. An unlimited number of active sessions is more clearly represented by a null value.
2. To match the semantics of the corresponding JBossWebMetaData property:
https://github.com/jboss/metadata/blob/master/web/src/main/java/org/jboss/metadata/web/jboss/JBossWebMetaData.java#L157